grouper/ctl/factory.py

```diff
@@ -0,0 +1,79 @@
+from typing import TYPE_CHECKING
+
+from grouper.ctl.permission import PermissionCommand
+from grouper.graph import Graph
+from grouper.models.base.session import get_db_engine, Session
+from grouper.repositories.factory import RepositoryFactory
+from grouper.services.factory import ServiceFactory
+from grouper.settings import settings
+from grouper.usecases.factory import UseCaseFactory
+from grouper.util import get_database_url
+
+if TYPE_CHECKING:
+    from argparse import _SubParsersAction
+    from grouper.ctl.base import CtlCommand
+    from grouper.graph import GroupGraph
+    from typing import Optional
+
+
+class UnknownCommand(Exception):
+    """Attempted to run a command with no known class."""
+
+    pass
+
+
+class CtlCommandFactory(object):
```

Comment:
It feels like this class has two responsibilities: building CtlCommand subclasses, and building top-level dependencies like Session, Graph, and UseCaseFactory. Not a big problem, but we should consider separating these responsibilities. The latter can be shared between API, FE, and CTL too.

Reply:
I'm going to merge this now so that I can merge some stuff blocked behind it, but will keep thinking about this and propose a fix in a follow-up PR.